pygmo/core.cpp

```diff
@@ -861,8 +869,87 @@ BOOST_PYTHON_MODULE(core)
 
     moead_.def("get_seed", &moead::get_seed);
 
+    // Exposition of various structured utilities
+    // Hypervolume class
+    bp::class_<hypervolume>("hypervolume", "Hypervolume Class")
+        .def("__init__", bp::make_constructor(+[](const population &pop) { return ::new hypervolume(pop, true); },
+                                              bp::default_call_policies(), (bp::arg("pop"))),
+             pygmo::hv_init2_docstring().c_str())
+        .def("__init__", bp::make_constructor(
+                             +[](const bp::object &points) { return ::new hypervolume(pygmo::to_vvd(points), true); },
```

Comment:
Calling ``new`` like this can lead to memory leaks, if for instance ``to_vvd()`` throws (which it might, if it is passed a wrong object from Python). ``to_vvd()`` must be called in its own line, and the resulting object can then be passed to the ``new`` expression constructing the hypervolume.